great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -4035,7 +3810,7 @@ def profile_data_asset(  # noqa: PLR0912, PLR0913, PLR0915
         )
 
         # TODO: Add batch_parameters
-        batch = self.get_batch(
+        batch = self._get_batch_v2(
```

Comment:
We've historically written pretty strongly coupled tests - we need this v2 behavior to maintain our tests but we don't actually want/allow users to run this specific logic.

The way I see it, all the tests that use this behavior are V2 and therefore are either deprecated or are up for deletion. I don't love it but I think keeping this private method for test scaffolding is appropriate.